### PR TITLE
Disable transpilation of contracts with @custom:stateless in peerProject mode

### PR DESCRIPTION
**Pull request by Amxx**
marking a contract as `/// @custom:stateless` will disable the transpilation of a contract when operating in peer project mode. These contracts will be treated as libraries or interfaces.

Note that marking a contract as stateless cannot be undone without introducing a breaking change.

Here are some contracts that I believe should be marked as stateless in `@openzeppelin/contracts`:
- Context
- ERC165
- Multicall
- ERC721Holder
- ERC1155Holder

### CODE REVIEW

**Comment by frangio** [comment]
Not sure about the `peer` part. What other modes do you have in mind?

If we go forward with that format, we should validate the arguments and reject anything that is not `peer`.

That said, I'm not sure I like the annotation name. Is it something we would want to have in the vanilla codebase, or only added in the upgradeable patch? I'm thinking that it will show up in vanilla verified source code otherwise, which is weird.

**Comment by Amxx** [comment]
> Not sure about the peer part. What other modes do you have in mind?

I'm thinking that in the future another mode could be used to exlucde a contract from the transpilation when `-q` is not enabled. Something like `-x` but in the source code.

**Comment by Amxx** [comment]
I like `@custom:stateless`.

File: src/utils/natspec.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export function* extractNatspec(node: {
     typeof node.documentation === 'string' ? node.documentation : node.documentation?.text ?? '';
 
   for (const { groups } of execall(
-    /^\s*(?:@(?<title>\w+)(?::(?<tag>[a-z][a-z-]*))? )?(?<args>(?:(?!^\s@\w+)[^])*)/m,
+    /^\s*(?:@(?<title>\w+)(?::(?<tag>[a-z][a-z-]*))?)?(?: (?<args>(?:(?!^\s@\w+)[^])*))?/m,
```
Comment (Amxx) [inline src/utils/natspec.ts:17]: FYI, the issue here is that the space in the tag block prevents parsing title+tag without a space/args at the end.

**Comment by Amxx** [comment]
Note: the last refactor removes many of the change from 0.3.28. In particular, exclude is once again a boolean (no more soft exclude). Soft exclusion is implemented by removing file from the Transform object after construction.

**Review by frangio — APPROVED** [review]

**Pull request merged**